packages/editor-ui/src/components/layouts/ResourcesListLayout.vue

```diff
@@ -28,54 +28,83 @@ export type IResource = {
 	sharedWithProjects?: ProjectSharingData[];
 };
 
-export interface IFilters {
+export type BaseFilters = {
 	search: string;
 	homeProject: string;
 	[key: string]: boolean | string | string[];
-}
+};
 
-type IResourceKeyType = 'credentials' | 'workflows' | 'variables';
+type ResourceKeyType = 'credentials' | 'workflows' | 'variables';
+
+const route = useRoute();
+const router = useRouter();
+const i18n = useI18n();
+const { callDebounced } = useDebounce();
+const usersStore = useUsersStore();
+const telemetry = useTelemetry();
 
 const props = withDefaults(
 	defineProps<{
-		resourceKey: IResourceKeyType;
-		displayName?: (resource: IResource) => string;
-		resources: IResource[];
+		resourceKey: ResourceKeyType;
+		displayName?: (resource: Resource) => string;
+		resources: Resource[];
 		disabled: boolean;
 		initialize?: () => Promise<void>;
-		filters?: IFilters;
+		filters?: BaseFilters;
 		additionalFiltersHandler?: (
-			resource: IResource,
-			filters: IFilters,
+			resource: Resource,
+			filters: BaseFilters,
 			matches: boolean,
 		) => boolean;
 		shareable?: boolean;
 		showFiltersDropdown?: boolean;
-		sortFns?: Record<string, (a: IResource, b: IResource) => number>;
+		sortFns?: Record<string, (a: Resource, b: Resource) => number>;
 		sortOptions?: string[];
-		type?: 'datatable' | 'list';
+		type?: 'datatable' | 'list-full' | 'list-paginated';
 		typeProps: { itemSize: number } | { columns: DatatableColumn[] };
 		loading: boolean;
+		customPageSize?: number;
+		availablePageSizeOptions?: number[];
+		totalItems?: number;
+		resourcesRefreshing?: boolean;
+		// Set to true if sorting and filtering is done outside of the component
+		dontPerformSortingAndFiltering?: boolean;
 	}>(),
 	{
-		displayName: (resource: IResource) => resource.name || '',
+		displayName: (resource: Resource) => resource.name || '',
 		initialize: async () => {},
 		filters: () => ({ search: '', homeProject: '' }),
 		sortFns: () => ({}),
 		sortOptions: () => ['lastUpdated', 'lastCreated', 'nameAsc', 'nameDesc'],
-		type: 'list',
+		type: 'list-full',
 		typeProps: () => ({ itemSize: 80 }),
 		loading: true,
 		additionalFiltersHandler: undefined,
 		showFiltersDropdown: true,
 		shareable: true,
+		customPageSize: 25,
+		availablePageSizeOptions: () => [10, 25, 50, 100],
+		totalItems: 0,
+		dontPerformSortingAndFiltering: false,
+		resourcesRefreshing: false,
 	},
 );
 
+const sortBy = ref(props.sortOptions[0]);
+const hasFilters = ref(false);
+const filtersModel = ref(props.filters);
```

Comment:
this will mutate the props, even if vue doesn't warn you about it. 
```suggestion
const filtersModel = ref(structuredClone(props.filters));
```

Reply:
I don't think this is true, `ref` is an independent reactive reference so it's just initialized to prop value but it shouldn't be double-bound to it.

Reply:
feel free to try it :)

if it where a string or a number, it wouldn't mutate, but since it's on object it will

Reply:
@MiloradFilipovic https://www.loom.com/share/4cb5adb3c1be413d85e18b83673e25d0

Reply:
Well, today I learned 😁 Just tried it in playground and looks like this is a way to break the one-way data flow that props should respect.
Ok, since we need this double binding, I'll implement it using `v-model` as it should be